hadoop-ozone/dev-support/checks/native_check.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function get_rocks_native_git_sha() {
+    echo "Getting Rocks Native Git sha"
+    echo
+    # Getting the latest git hash for the ./hadoop-hdds/rocks-native directory
+    echo "git log -n 1 --format=\"%h\" ./hadoop-hdds/rocks-native"
+    ROCKS_NATIVE_GIT_SHA=$(git log -n 1 --format="%h" ./hadoop-hdds/rocks-native)
+    echo "ROCKS_NATIVE_GIT_SHA = ${ROCKS_NATIVE_GIT_SHA}"
+    readonly ROCKS_NATIVE_GIT_SHA
+}
+
+function init_native_maven_opts() {
+    get_rocks_native_git_sha
+    PROJECT_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+    # Parsing out version number from project version by getting the first occurance of '-'.
+    # If project version is 1.4.0-SNAPSHOT, VERSION_NUMBER = 1.4.0
+    VERSION_NUMBER=$(echo "${PROJECT_VERSION}" | cut -f1 -d'-')
+    # Adding rocks native sha after the version number in the project version.
+    # EXPECTED_ROCK_NATIVE_VERSION = 1.4.0.<rocks native git sha>-SNAPSHOT
```

Comment:
@swamirishi @smengcl Instead of `.` please use `-` to separate version number and hash to avoid failure in _dependency_ check:

```
-share/ozone/lib/hdds-rocks-native.jar
+share/ozone/lib/hdds-rocks-nativecd91-SNAPSHOT.jar
```

Dependency check passes after:

```
mv -iv hadoop-ozone/dist/target/ozone-1.4.0-SNAPSHOT/share/ozone/lib/hdds-rocks-native-1.4.0.646cd91-SNAPSHOT.jar hadoop-ozone/dist/target/ozone-1.4.0-SNAPSHOT/share/ozone/lib/hdds-rocks-native-1.4.0-646cd91-SNAPSHOT.jar
```